### PR TITLE
Add ability to set the expiry time of a secret

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Add an extra param to the API to pass in the expiry value in seconds.

And also add `secret` to the filtered params list so we never log that info, even if it's encrypted.

### CODE REVIEW

File: api/lib/secrets_api/secrets.ex [hunk 1]
```diff
@@ -8,12 +8,12 @@ defmodule SecretsApi.Secrets do
...
 
-  @spec store_secret(any) ::
+  @spec store_secret(any, any) ::
           {:error, charlist()} | {:ok, binary}
-  def store_secret(secret) do
+  def store_secret(secret, expiry \\ 3600) do
     room_id = generate_room_id()
 
-    case Redix.command(["SET", room_id, secret, "EX", "3600", "NX"]) do
+    case Redix.command(["SET", room_id, secret, "EX", expiry, "NX"]) do
```
Comment (ZePedroResende) [inline api/lib/secrets_api/secrets.ex:16]: we should probably check somewhere that this expiry is in fact one of the values that we let the users send on the frontend so we dont have infinite time expiry secrets

Comment (ZePedroResende) [inline api/lib/secrets_api/secrets.ex:16]: we should probably check somewhere that this expiry is in fact one of the values that we let the users send on the frontend so we dont have infinite time expiry secrets

Comment (jfranciscosousa) [reply api/lib/secrets_api/secrets.ex:16]: Given it's an integer, I'll just set a max allowed value

Comment (ZePedroResende) [reply api/lib/secrets_api/secrets.ex:16]: Sure ! seems a good solution

**Review by ZePedroResende — APPROVED** [review]

**Pull request merged**